mcp/src/rustic_ai/mcp/agent.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+from contextlib import AsyncExitStack
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.sse import sse_client
+from mcp.client.stdio import stdio_client
+from rustic_ai.core.guild import agent
+from rustic_ai.core.guild.agent import Agent, ProcessContext
+from rustic_ai.core.guild.agent_ext.mixins.guild_refresher import GuildRefreshMixin
+
+from .models import (
+    CallToolRequest,
+    CallToolResponse,
+    MCPAgentConfig,
+    MCPClientType,
+    MCPServerConfig,
+    ToolResult,
+)
+
+
+@dataclass
+class MCPSessionContext:
+    session: ClientSession
+    exit_stack: AsyncExitStack
+    client_type: str
+
+
+class MCPAgent(Agent[MCPAgentConfig], GuildRefreshMixin):
+    """
+    Agent that connects to MCP servers and exposes their capabilities.
+
+    This agent manages MCP client sessions. Due to asyncio event loop constraints,
+    sessions must be created and used within the same event loop context.
+    """
+
+    def __init__(self):
+        self._sessions: Dict[str, MCPSessionContext] = {}
```

Comment:
One agent should only connect to one single MCP server and have 1 session,  what would be the use case for multiple sessions?